### PR TITLE
[CssSelector] remove ConverterInterface

### PR DESCRIPTION
**Pull request by fabpot**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #15934 |
| License | MIT |
| Doc PR | n/a |

### CODE REVIEW

**Comment by fabpot** [comment]
The `Converter` name is probably not the best one. I'm not a big fan of `CssToXPathConverter` either and I think the best one is still `CssSelector` (which implements the static interface). Not sure what to do here. ping @symfony/deciders

**Comment by xabbuh** [comment]
:+1: 

Status: Reviewed

About the class name: We could merge the `Converter` and the `CssSelector` class and only deprecate the static method.

File: src/Symfony/Component/CssSelector/Converter.php [hunk 1]
```diff
@@ -19,11 +19,48 @@
...
 
 /**
- * @author Christophe Coevoet <stof@notk.org>
+ * Converter is the main entry point of the component and can convert CSS
+ * selectors to XPath expressions.
+ *
+ * This component is a port of the Python cssselect library,
+ * which is copyright Ian Bicking, @see https://github.com/SimonSapin/cssselect.
+ *
+ * Copyright (c) 2007-2012 Ian Bicking and contributors. See AUTHORS
```
Comment (stof) [inline src/Symfony/Component/CssSelector/Converter.php]: @fabpot is the phpdoc the right place to put the license of the ported library ? This class is not even ported from the Python library (the classes porting the Pythin library are all the internal classes)

Comment (fabpot) [inline src/Symfony/Component/CssSelector/Converter.php]: moved to the README file instead

**Comment by fabpot** [comment]
@xabbuh `toXPath()` are in both the new and the old class (one is static, the other one is not).

File: src/Symfony/Component/CssSelector/Converter.php [hunk 2]
```diff
@@ -19,11 +19,12 @@
...
 /**
- * @author Christophe Coevoet <stof@notk.org>
+ * Converter is the main entry point of the component and can convert CSS
+ * selectors to XPath expressions.
  *
- * @api
+ * @author Christophe Coevoet <stof@notk.org>
  */
-class Converter implements ConverterInterface
+class Converter
```
Comment (stloyd) [inline src/Symfony/Component/CssSelector/Converter.php]: Maybe this should be called `Transformer`? As this transforms one convention to another ;)

**Comment by xabbuh** [comment]
@fabpot We could rename the new method to `asXPath()` or something like that or choose a class name like `CssTransformer`.

**Comment by Tobion** [comment]
I think `CssSelector` would be the wrong name for a non-static class. In OOP I would image an API of a CssSelector like `(new CssSelector($css))->toXPath()`. So the CssSelector represents a single css expression. But that is not the API we have here right now. So Converter/Transformer (but more specific) makes alot more sense, as you only need one converter to transform alot of css expressions.

**Comment by weaverryan** [comment]
How about any of `CssConverter`, `CssTransformer` or `CssTranslator`?

**Comment by jakzal** [comment]
or `CssSelectorConverter` ?

**Comment by Tobion** [comment]
+1 for `CssSelectorConverter ` as we are converting a selector and not css code

**Comment by fabpot** [comment]
Renamed and fixed the README to use the new API.

**Comment by xabbuh** [comment]
:+1:

**Comment by stof** [comment]
This is broken. You forgot to use the new class name in the DomCrawler component (why do we have a red flag on Travis and AppVeyor if nobody look at the test failures ?)

**Comment by fabpot** [comment]
Indeed, sorry about that, doing a PR right now.

**Comment by fabpot** [comment]
see #16012

**Comment by xabbuh** [comment]
We need to make sure that our testsuite gets back to green asap so that we spot things like this faster. :/

**Pull request merged**